runtime-modules/working-team/src/benchmarking.rs

```diff
@@ -0,0 +1,921 @@
+        &caller_id,
+        &member_id,
+        &JobOpeningType::Leader
+        );
+
+    }: _ (RawOrigin::Signed(caller_id.clone()), application_id)
+    verify {
+      assert!(!ApplicationById::<T, I>::contains_key(application_id), "Application not removed");
+      assert_last_event::<T, I>(RawEvent::ApplicationWithdrawn(application_id).into());
+    }
+
+    // Regular worker is the worst case scenario since the checks
+    // require access to the storage whilist that's not the case with a lead opening
+    slash_stake {
+      let i in 0 .. 10;
+
+      let (lead_id, lead_worker_id) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+      let (caller_id, worker_id) = insert_a_worker::<T, I>(StakingRole::WithStakes, JobOpeningType::Regular, 1, Some(lead_id.clone()));
+      let slashing_amount = One::one();
+      let penalty = Penalty {
+        slashing_text: vec![],
+        slashing_amount,
+      };
+    }: _(RawOrigin::Signed(lead_id.clone()), worker_id, penalty)
+    verify {
+      assert_last_event::<T, I>(RawEvent::StakeSlashed(worker_id, slashing_amount).into());
+    }
+
+    terminate_role_worker {
+      let i in 0 .. 10;
+
+      let (lead_id, _) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+      let (caller_id, worker_id) = insert_a_worker::<T, I>(StakingRole::WithStakes, JobOpeningType::Regular, 1, Some(lead_id.clone()));
+      // To be able to pay unpaid reward
+      let current_budget = BalanceOfCurrency::<T>::max_value();
+      WorkingTeam::<T, _>::set_budget(RawOrigin::Root.into(), current_budget).unwrap();
+      let penalty = Penalty {
+        slashing_text: vec![],
+        slashing_amount: One::one(),
+      };
+    }: terminate_role(RawOrigin::Signed(lead_id.clone()), worker_id, Some(penalty))
+    verify {
+      assert!(!WorkerById::<T, I>::contains_key(worker_id), "Worker not terminated");
+      assert_last_event::<T, I>(RawEvent::TerminatedWorker(worker_id).into());
+    }
+
+    terminate_role_lead {
+      let i in 0 .. 10;
+
+      let (_, lead_worker_id) = insert_a_worker::<T, I>(StakingRole::WithStakes, JobOpeningType::Leader, 0, None);
+      let current_budget = BalanceOfCurrency::<T>::max_value();
+      // To be able to pay unpaid reward
+      WorkingTeam::<T, _>::set_budget(RawOrigin::Root.into(), current_budget).unwrap();
+      let penalty = Penalty {
+        slashing_text: vec![],
+        slashing_amount: One::one(),
+      };
+    }: terminate_role(RawOrigin::Root, lead_worker_id, Some(penalty))
+    verify {
+      assert!(!WorkerById::<T, I>::contains_key(lead_worker_id), "Worker not terminated");
+      assert_last_event::<T, I>(RawEvent::TerminatedLeader(lead_worker_id).into());
+    }
+
+    // Regular worker is the worst case scenario since the checks
+    // require access to the storage whilist that's not the case with a lead opening
+    increase_stake {
+      let i in 0 .. 10;
```

Comment:
For stake related 'verify' functions, you could check the actual change of the stake.

Reply:
I think a problem with that it's that it depends on the specific implementation of `StakingHandler` in the runtime or in the verification test in the `Test`.

In fact, I think we can't check for `current_stake` since that (almost)always returns 100  in the current mock implementation. I'm not sure but maybe we can for the locked amount?

Reply:
I agree. Let's skip it then.